src/Core/Injector/Injector.php

```diff
@@ -876,8 +909,10 @@ public function unregisterAllObjects()
      */
     public function get($name, $asSingleton = true, $constructorArgs = null)
     {
+        $object = null;
```

Comment:
I'm not too happy with all the `if(!$object)` conditionals, but the only alternative I can see is extracting the `get()` implementation into a new protected method which is then called by a much leaner `get()`